feat: Add no-unknown-wire-adapters linting rule (#29)
salesforce/eslint-plugin-lwc: docs/rules/no-unknown-wire-adapters.md

## Disallow usage of unknown wire adapters (no-unknown-wire-adapters)

> This rule offers a way to restrict the usage of unexpected wire adapters. It can be used to ensure that all the LWC components wired properties are statically analyzable (for example for prefetching or offline purposes).

Reviewer: ```suggestion
This rule offers a way to restrict the usage of unknown wire adapters. It can be used to ensure that all wired properties are statically analyzable in order to support use cases such as prefetching and offline.
```

## Rule details

> This rule can be configured as followed:

Reviewer 2: ```suggestion
This rule can be configured as follows:
```

```js
{
    adapters: [
        // for modules exporting wire adapters using named exports
        { module: '<module name>', identifier: '<adapter identifier>' },

        // for modules exporting wire adapters using default exports
        { module: '<module name>', identifier: 'default' },
    ];
}
```

> By default the list of known adapters is empty. So all the `@wire` decorator usages will be flagged as an error.

Reviewer: ```suggestion
Any usage of the `@wire` decorator will be flagged as an error unless a list of known adapters has been provided.
```
